### ISSUE TITLE
Collect a list of functions that do not yet use regions

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#3841]
Collect a list of functions that do not yet use regions, either because they were overlooked or because they currently crash the compiler. Include the ticket when relevant.

**Comment by magnus-madsen** [comment]
@jaschdoc If you could start compiling such a list that would be useful :)

**Comment by jaschdoc** [comment]
# WIP

| Namespace / File | Function                      | Ticket? |
| `Array`          | `indices`                     |         |
| `Array`          | `flatMap`                     |         |
| `Array`          | `intercalate`                 |         |
| `Benchmark`      | `runWithBudget`               |         |
| `Benchmark`      | `warmupAndEstimate`           |         |
| `Benchmark`      | `runAll`                      |         |
| `Environment`    | `getArgs`                     |         |
| `Random`         | `choose`                      |         |
| `String`         | `split`                       |         |
| `String`         | `arraySwap`                   |         |
| `String`         | `zip`                         |         |
| `Bench.Chain`    | `listOfArray`                 |         |
| `Bench.Chain`    | `chainOfArray`                |         |
| `Bench.Chain`    | `listCons`                    |         |
| `Bench.Chain`    | `chainCons`                   |         |
| `Bench.Chain`    | `listSnoc`                    |         |
| `Bench.Chain`    | `chainSnoc`                   |         |
| `Bench.Chain`    | `listAppend`                  |         |
| `Bench.Chain`    | `chainAppend`                 |         |
| `Bench.Chain`    | `chainSuite`                  |         |
| `Bench.Fixpoint` | `benchmarks`                  |         |
| `Bench.Fixpoint` | `benchmarkProjectInto`        |         |
| `Bench.Fixpoint` | `benchmarkSelect`             |         |
| `Bench.Fixpoint` | `benchmarkSolve`              |         |
| `Bench.Fixpoint` | `benchmarkCrossProductSolve2` |         |
| `Bench.Fixpoint` | `benchmarkCrossProductSolve3` |         |
| `Bench.Fixpoint` | `benchmarkSolveIntersect`     |         |
| `Bench.Fixpoint` | `benchmarkCompile`            |         |

**Comment by magnus-madsen** [comment]
@jaschdoc Once #3876 is merged, could you teach @ninaandrup how to refactor `Benchmark.flix` to use regions (1st PR) and then subsequently refactor the benchmarks (2nd pr)?

**Comment by magnus-madsen** [comment]
The `File` namespace has to be impure, so they can be removed from your list.

**Comment by magnus-madsen** [comment]
For `String` I think you should see if they can be refactored. For `Array` we know these functions are broken (crash the compiler), but try to see if you can make it work by introducing top-level helper functions. That might help compared to local defs.

**Comment by magnus-madsen** [comment]
@jaschdoc Could I ask you to update the table at some point? (Just delete the items no longer relevant).

**Comment by jaschdoc** [comment]
> @jaschdoc Could I ask you to update the table at some point? (Just delete the items no longer relevant).

Yeah sure. I still haven't gotten around to finishing it though, but will do that soon :)

**Comment by jaschdoc** [comment]
The list should be up to date and finished now!
